File: test_api.py
import requests
import json

url = "https://script.google.com/macros/s/AKfycbwgdwkHnJcPURZerkrPkhbJUuGFb1qTOWEXft1OJQrphR36DVGeWVvzAUaZs_MPG4-R/exec?action=get_registration&empId=ops66909"
response = requests.get(url)

try:
    data = response.json()
    print("Fetched data length:", len(data))
    for d in data:
        print(f"Shift ID: {d.get('shiftId')}, Shift Label: {d.get('shiftLabel')}")
        print("Selections:", len(d.get('selections', [])))
except Exception as e:
    print("Error:", e)
    print("Response text:", response.text)

File: test_api.js
const fetch = require('node-fetch');

async function testReg() {
  const payload = {
    action: 'get_registration',
    empId: process.argv[2] || 'ops236220'
  };

  const apiLink = 'https://script.google.com/macros/s/AKfycbyfXco9SQzgQBFOucTUzRd9tRX4MqsogCBLi0ANnNDOiH7KG2e6itEu3bioGUjtnbtw/exec';
  
  try {
    const res = await fetch(apiLink, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(payload)
    });
    console.log("Status:", res.status);
    const text = await res.text();
    console.log("Response:", text);
  } catch (err) {
    console.error("Error:", err);
  }
}

testReg();
